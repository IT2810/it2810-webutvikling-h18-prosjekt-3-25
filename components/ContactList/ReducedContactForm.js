import React from 'react';
import { StyleSheet, Text, View, TouchableHighlight } from 'react-native';
import { Avatar, Icon } from 'react-native-elements';

export default class ReducedContactForm extends React.Component {
    render() {
        return (
            //TouchableHighlight gjør elementet trykkbart og lager effekten
            //man ser når man trykker på det
            <TouchableHighlight onPress={this.props.onPress}>
                <View style={styles.contactFormContainer}>
                    <Avatar
                        medium
                        rounded
                        title={this.props.name.split(" ").map(x => x[0]).join("")}
                    />
                    <View style={styles.names}>
                        <Text style={styles.text}>{this.props.name}</Text>
                    </View>
                </View>
            </TouchableHighlight>
        );
    }
}

const styles = StyleSheet.create({

    contactFormContainer: {
        flex: 1,
        flexDirection: "row",
        paddingTop: 10,
        paddingBottom: 10,
        borderBottomWidth: 1,
        borderBottomColor: "#aaa",
    },

    names: {
        justifyContent: "center",
    },

    text: {
        paddingLeft: 20,
        fontSize: 18,
    }
});
