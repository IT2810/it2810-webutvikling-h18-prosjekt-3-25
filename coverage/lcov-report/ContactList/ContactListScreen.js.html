<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for ContactList/ContactListScreen.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">ContactList</a> ContactListScreen.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.07% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>18/29</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">64% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>16/25</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { StyleSheet, Text, View, FlatList, AsyncStorage } from 'react-native';
import { Avatar, Icon } from 'react-native-elements';
import ContactScreen from '../Contact/ContactScreen.js';
import ReducedContactForm from './ReducedContactForm'
&nbsp;
export default class ContactList extends React.Component {
    //Her settes tittelen som brukes i navigasjonsbaren
    static navigationOptions = {
        title: "ContactList",
    }
&nbsp;
    constructor(props) {
        super(props);
        this.state = {
            contacts: this.retrieveData(),
        };
&nbsp;
        this.handleOnPress = this.handleOnPress.bind(this);
    }
&nbsp;
    // Henter kontakt-dataen som er lagret ved AsyncStorage og oppdaterer state
    async retrieveData() {
        try {
            const value = await AsyncStorage.getItem("CONTACTS");
            if (value !== undefined) {
                const parsedValue = JSON.parse(value);
                this.setState(state =&gt; ({contacts: parsedValue}));
                return parsedValue;
            } else {
                this.setState(state =&gt; ({contacts: []}));
                return [];
            }
        } catch(error) {
<span class="cstat-no" title="statement not covered" >            console.log(error);</span>
<span class="cstat-no" title="statement not covered" >            return []</span>
        }
    }
&nbsp;
    // Lagrer listen med kontakter, contactList, i AsyncStorage
    async storeData(contactList) {
        const contactListString = JSON.stringify(contactList);
        try {
            await AsyncStorage.setItem("CONTACTS", contactListString);
        } catch(error) {
<span class="cstat-no" title="statement not covered" >            console.log(error);</span>
        }
    }
&nbsp;
    // Kalles n√•r en ny kontakt blir opprettet
<span class="fstat-no" title="function not covered" >    ha</span>ndleOnPress(contact) {
        let stateClone = <span class="cstat-no" title="statement not covered" >[...this.state.contacts, contact];</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >        this.storeData(stateClone);</span>
<span class="cstat-no" title="statement not covered" >        this.setState(<span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{</span></span>
            contacts: stateClone,
        }))
    }
&nbsp;
    render() {
        return (
            &lt;View style={styles.contactListContainer}&gt;
&nbsp;
                &lt;View style={styles.addContact}&gt;
                    &lt;Icon
                        name="person-add"
                        size={70}
                        onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >this.props.navigation.navigate("CreateContact", {onPress: this.handleOnPress})}</span>
                    /&gt;
                &lt;/View&gt;
&nbsp;
                &lt;View style={styles.listContainter}&gt;
                    &lt;Text&gt;Contacts&lt;/Text&gt;
                    &lt;FlatList
                        data={this.state.contacts}
                        keyExtractor={<span class="fstat-no" title="function not covered" >(i</span>tem, index) =&gt; <span class="cstat-no" title="statement not covered" >item.number}</span>
                        // rendrer kontaktene i listen, og navigeter til kontakten med name, number og email som props n√•r man trykker p√• kontakten
                        renderItem={<span class="fstat-no" title="function not covered" >({</span>item}) =&gt; <span class="cstat-no" title="statement not covered" >&lt;ReducedContactForm name={item.name} onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >this.props.navigation.navigate("Contact", {name: item.name, number: item.number, email: item.email})}</span>/&gt;}</span>
                    /&gt;
                &lt;/View&gt;
&nbsp;
            &lt;/View&gt;
        );
    }
}
&nbsp;
const styles = StyleSheet.create({
    contactListContainer: {
        flex: 1,
    },
&nbsp;
    addContact: {
        alignItems: "flex-end",
    },
&nbsp;
    listContainter: {
        flex: 1,
    },
});
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 19 2018 12:44:31 GMT+0200 (Vest-Europa (sommertid))
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
