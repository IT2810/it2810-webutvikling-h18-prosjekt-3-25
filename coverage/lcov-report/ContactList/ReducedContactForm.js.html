<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for ContactList/ReducedContactForm.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">ContactList</a> ReducedContactForm.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { StyleSheet, Text, View, TouchableHighlight } from 'react-native';
import { Avatar, Icon } from 'react-native-elements';
&nbsp;
export default class ReducedContactForm extends React.Component {
    render() {
        return (
            //TouchableHighlight gjør elementet trykkbart og lager effekten
            //man ser når man trykker på det
            &lt;TouchableHighlight onPress={this.props.onPress}&gt;
                &lt;View style={styles.contactFormContainer}&gt;
                    &lt;Avatar
                        medium
                        rounded
                        title={this.props.name.split(" ").map(x =&gt; x[0]).join("")}
                    /&gt;
                    &lt;View style={styles.names}&gt;
                        &lt;Text style={styles.text}&gt;{this.props.name}&lt;/Text&gt;
                    &lt;/View&gt;
                &lt;/View&gt;
            &lt;/TouchableHighlight&gt;
        );
    }
}
&nbsp;
const styles = StyleSheet.create({
&nbsp;
    contactFormContainer: {
        flex: 1,
        flexDirection: "row",
        paddingTop: 10,
        paddingBottom: 10,
        borderBottomWidth: 1,
        borderBottomColor: "#aaa",
    },
&nbsp;
    names: {
        justifyContent: "center",
    },
&nbsp;
    text: {
        paddingLeft: 20,
        fontSize: 18,
    }
});
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 19 2018 12:44:31 GMT+0200 (Vest-Europa (sommertid))
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
