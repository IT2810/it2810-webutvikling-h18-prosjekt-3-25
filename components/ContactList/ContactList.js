import React from 'react';
import { StyleSheet, Text, View, FlatList } from 'react-native';
import { Avatar, Icon } from 'react-native-elements';
import Contact from '../Contact/Contact.js';
import ReducedContactForm from './ReducedContactForm'

export default class ContactList extends React.Component {
    render() {
        return (
            <View style={styles.contactListContainer}>
                <View style={styles.addContact}>
                    <Icon
                        name="person-add"
                        size={70}
                        onPress={() => this.props.navigation.navigate("CreateContactScreen")}
                    />
                </View>
                <View style={styles.listContainter}>
                    <Text>Contacts</Text>
                    <FlatList
                        data={[
                            {
                                key: "Leif",
                                name: "Leif Ulvund",
                                number:"45432377",
                                email: "leif.ulvund@gmail.com",
                            },
                            {
                                key: "Manu",
                                name: "Emanuele Caprioli",
                                number: "92026636",
                                email: "manu.caprioli@gmail.com",
                            },
                        ]}
                        renderItem={({item}) => <ReducedContactForm name={item.name}/>}
                    />
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    contactListContainer: {
        flex: 1,
    },

    addContact: {
        // flex: 1,
        alignItems: "flex-end",
    },

    listContainter: {
        flex: 1,
    },
});
