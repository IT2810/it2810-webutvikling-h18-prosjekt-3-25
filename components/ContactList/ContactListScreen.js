import React from 'react';
import { StyleSheet, Text, View, FlatList } from 'react-native';
import { Avatar, Icon } from 'react-native-elements';
import ContactScreen from '../Contact/ContactScreen.js';
import ReducedContactForm from './ReducedContactForm'

export default class ContactList extends React.Component {
    //Her settes tittelen som brukes i navigasjonsbaren
    static navigationOptions = {
        title: "ContactList",
    }

    constructor(props) {
        super(props);

        this.state = {
            users: [
                {
                    name: "Leif Ulvund",
                    number: "45432377",
                    email: "leif.ulvund@gmail.com",
                },
                {
                    name: "Emanuele Caprioli",
                    number: "92026636",
                    email: "manu.caprioli@gmail.com",
                },
            ],
        };

        this.handleOnPress = this.handleOnPress.bind(this);
    }

    handleOnPress(contact) {
        let stateClone = [...this.state.users, contact]
        this.setState(state => ({
            users: stateClone,
        }))
    }

    render() {
        return (
            <View style={styles.contactListContainer}>

                <View style={styles.addContact}>
                    <Icon
                        name="person-add"
                        size={70}
                        onPress={() => this.props.navigation.navigate("CreateContact", {onPress: this.handleOnPress})}
                    />
                </View>

                <View style={styles.listContainter}>
                    <Text>Contacts</Text>
                    <FlatList
                        data={this.state.users}
                        keyExtractor={(item, index) => item.number}
                        // rendrer kontaktene i listen, og navigeter til kontakten med name, number og email som props når man trykker på kontakten
                        renderItem={({item}) => <ReducedContactForm name={item.name} onPress={() => this.props.navigation.navigate("Contact", {name: item.name, number: item.number, email: item.email})}/>}
                    />
                </View>

            </View>
        );
    }
}

const styles = StyleSheet.create({
    contactListContainer: {
        flex: 1,
    },

    addContact: {
        // flex: 1,
        alignItems: "flex-end",
    },

    listContainter: {
        flex: 1,
    },
});
