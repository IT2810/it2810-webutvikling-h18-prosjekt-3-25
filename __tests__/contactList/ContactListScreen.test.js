import React from 'react';
import { AsyncStorage } from 'react-native';
import ContactList from '../../components/ContactList/ContactListScreen.js';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

// const renderer = new ShallowRenderer();
// renderer.render(<ContactList />);
// const result = renderer.getRenderOutput();
// console.log(result.props.children);
// console.log(result.type);

// jest.mock('react-native', () => {
//     AsyncStorage: {
//         setItem: jest.fn(() => {
//             return new Promise((resolve, reject) => {
//                 resolve(null);
//             })
//         })
//
//         getItem: jest.fn(() => {
//             return new Promise((resolve, reject) => {
//                 resolve({
//                     name: "Test Person",
//                     number: "12345678",
//                     email: "email@email.com",
//                 });
//             })
//         })
//     }
// })
//
// jest.mock('../../components/ContactList/ContactListScreen.js', () => {
//     ContactList: {
//         retrieveData: jest.fn(() => {
//             return {
//                 name: "Test Person",
//                 number: "12345678",
//                 email: "email@email.com",
//             }
//         })
//
//         storeData: jest.fn(() => {
//             return new Promise((resolve, reject) => {
//                 resolve(null);
//             })
//         })
//     }
// })

const contact = {
    name: "Test Person",
    number: "12345678",
    email: "email@email.com",
};

jest.mock('../../components/ContactList/ContactListScreen.js');

//Before all
const renderedComponent = renderer.create(<ContactList />);
const componentInstance = renderedComponent.getInstance()

it('renders correctly', () => {
    tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
});

// it("handleOnPress works correctly", async () => {
    // const contact = {
    //     name: "Test Person",
    //     number: "12345678",
    //     email: "email@email.com",
    // };
//
//     expect(componentInstance.state.contacts).toEqual([]);
//     await componentInstance.handleOnPress(contact);
//     console.log("etter state: " + componentInstance.state.contacts);
// })


it('retrieveData works correctly with no contacts', () => {
    const result = componentInstance.retrieveData().mockResolvedValue(contact);
    expect.assertions(2);
    return componentInstance.retrieveData().then(data => {
        // expect(componentInstance.state.contacts).toEqual([]);
        expect(data).toEqual(contact);
    })
});

// it('storeData works correctly', async () => {
//     const contact = {
//         name: "Test Person",
//         number: "12345678",
//         email: "email@email.com",
//     }
//
//
//     await componentInstance.storeData(contact);
//     return expect(componentInstance.state.contacts).toEqual([{name: "Test Person", number: "12345678", email: "email@email.com" }]);
//
//     // return componentInstance.storeData(contact).then(() => {
//     //     return componentInstance.retrieveData().then(() => {
//     //         console.log("state: " + JSON.stringify(componentInstance.state.contacts));
//     //         expect(componentInstance.state.contacts).toEqual([{
//     //             name: "Test Person",
//     //             number: "12345678",
//     //             email: "email@email.com",
//     //         }]);
//     //     })
//     // })
// })

// it('retrieveData works correctly with contacts', () => {
//     expect.assertions(1);
//
// })
