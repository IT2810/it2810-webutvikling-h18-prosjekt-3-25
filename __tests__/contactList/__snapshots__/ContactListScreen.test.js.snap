// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 70,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        îŸ¾
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Contacts
    </Text>
  </View>
</View>
`;
